### PR TITLE
[Fix](inverted index) add DATEV2 and DATETIMEV2 for inverted index reader

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

This PR adds support for DATEV2 and DATETIMEV2 types in the inverted index reader, aligning with the types already allowed in the inverted index writer.

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/inverted_index_reader.h [hunk 1]
```diff
@@ -407,6 +407,8 @@ class InvertedIndexQueryParamFactory {
             M(PrimitiveType::TYPE_CHAR)
             M(PrimitiveType::TYPE_VARCHAR)
             M(PrimitiveType::TYPE_STRING)
+            M(PrimitiveType::TYPE_DATEV2)
+            M(PrimitiveType::TYPE_DATETIMEV2)
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_reader.h:411]: add testcase for TYPE_DATEV2 and TYPE_DATETIMEV2

Comment (airborne12) [reply be/src/olap/rowset/segment_v2/inverted_index_reader.h:411]: already have

**Review by csun5285 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]

**Pull request merged**